app/src/main/java/tech/relaycorp/gateway/domain/courier/GenerateCCA.kt

```diff
@@ -9,14 +9,16 @@ import kotlin.time.days
 class GenerateCCA
 @Inject constructor(
     private val publicGatewayPreferences: PublicGatewayPreferences,
-    private val localConfig: LocalConfig
+    private val localConfig: LocalConfig,
+    private val calculateCreationDate: CalculateCRCMessageCreationDate
 ) {
 
     suspend fun generate() =
         CargoCollectionAuthorization(
             recipientAddress = publicGatewayPreferences.getCogRPCAddress(),
             payload = "".toByteArray(),
             senderCertificate = localConfig.getCertificate(),
+            creationDate = calculateCreationDate.calculate(),
             ttl = TTL.inSeconds.toInt()
```

Comment:
I missed this in the first version of this PR: The TTL should be relative to the `creationDate`, which is going to be particularly important with this PR since the creation date could be the start date of the cert.

We solved this in the cargo generation by using `Duration.between()`, so we should probably factor that out like you've done with `CalculateCRCMessageCreationDate` -- perhaps re-purposing that class to add a second method like `calculateTTL()`.

Reply:
But it would be 5 minutes ago or sooner, while the TTL is 14 days. It won't matter much would it?

Reply:
You're right that with the current version of the code it won't make any difference. However, I'm planning to make a change at some point before the stable release that could make a difference:

I'm planning to change all the dates on cargoes and CCAs so that they're totally inaccurate, and this way an eavesdropper couldn't infer at what time the user and the courier met. The meeting time could be used to infer location. For this reason, the plan is to set the creation date to a random number of days in the past. I haven't yet figured out how big to make this range.

Reply:
Then the TTL should also have some randomness to it I imagine. But I can make the TTL calculated for now as well.

Reply:
Good point! Yeah, the TTL should be the sum of the random duration in the past + the actual TTL. So this future change will be more elaborate than I thought anyway, so let's leave this as it is.